# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/typeaheadfind/nsTypeAheadFind.cpp
# Commit: 21cb609fe669
# Full Hash: 21cb609fe6690fbad6c6167e10ee2e7f9804a593
# Author: David Zbarsky <dzbarsky@gmail.com>
# Date: 2012-10-16 08:09:24
# Regressor Bug: 779684
# File Overlap Count: 1
# Description:
#   Backout bug 779684 for topcrash
# ==============================================================================

diff -r a96deba7de41 -r 21cb609fe669 toolkit/components/typeaheadfind/nsTypeAheadFind.cpp
--- a/toolkit/components/typeaheadfind/nsTypeAheadFind.cpp	Mon Oct 15 01:42:40 2012 -0400
+++ b/toolkit/components/typeaheadfind/nsTypeAheadFind.cpp	Mon Oct 15 14:35:50 2012 -0400
@@ -25,7 +25,7 @@
 #include "nsIDOMNode.h"
 #include "mozilla/dom/Element.h"
 #include "nsIFrame.h"
-#include "nsFrameIterator.h"
+#include "nsFrameTraversal.h"
 #include "nsIImageDocument.h"
 #include "nsIDOMHTMLDocument.h"
 #include "nsIDOMHTMLElement.h"
@@ -63,6 +63,8 @@
 NS_IMPL_ADDREF(nsTypeAheadFind)
 NS_IMPL_RELEASE(nsTypeAheadFind)
 
+static NS_DEFINE_CID(kFrameTraversalCID, NS_FRAMETRAVERSAL_CID);
+
 #define NS_FIND_CONTRACTID "@mozilla.org/embedcomp/rangefind;1"
 
 nsTypeAheadFind::nsTypeAheadFind():
@@ -1112,12 +1114,23 @@
   // We know that the target range isn't usable because it's not in the
   // view port. Move range forward to first visible point,
   // this speeds us up a lot in long documents
-  nsFrameIterator frameTraversal(aPresContext, frame,
-                                 eLeaf, nsFrameIterator::FLAG_NONE);
+  nsCOMPtr<nsIFrameEnumerator> frameTraversal;
+  nsCOMPtr<nsIFrameTraversal> trav(do_CreateInstance(kFrameTraversalCID));
+  if (trav)
+    trav->NewFrameTraversal(getter_AddRefs(frameTraversal),
+                            aPresContext, frame,
+                            eLeaf,
+                            false, // aVisual
+                            false, // aLockInScrollView
+                            false     // aFollowOOFs
+                            );
+
+  if (!frameTraversal)
+    return false;
 
   while (rectVisibility == nsRectVisibility_kAboveViewport) {
-    frameTraversal.Next();
-    frame = frameTraversal.CurrentItem();
+    frameTraversal->Next();
+    frame = frameTraversal->CurrentItem();
     if (!frame)
       return false;
 
